src/record/io.rs

```diff
@@ -175,6 +175,14 @@ impl<'a> Iterator for PartIndexIter<'a> {
             None
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        if self.num_points < 0 {
+            (0, None)
+        } else {
+            (self.num_points as usize, Some(self.num_points as usize))
```

Comment:
Just a drive by review since I'm not familiar with the internals of the shapefile crate, but size_hint should reflect the *remaining* elements in the iterator. 

See https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint

E.g. if I iterate through the first 30 points, the size_hint should be 30 less than it was before.

Does this code account for that? I don't see where num_points is ever decremented. Maybe we need to include current_part_index in the calculation?

Reply:
Thanks for the note about the remaining length.  I've committed an update that uses the position in the parts index for the part iterator for the hint.  The main shape iterator already used the remaining length via the shape index iterator.

Reply:
Thanks for the review